packages/mono-repo-publish/src/main.ts

```diff
@@ -12,74 +12,107 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-const { Octokit } = require('@octokit/rest');
-const { createAppAuth } = require('@octokit/auth-app');
-const fs = require('fs');
-const { readFileSync } = fs;
-const { dirname, join, resolve } = require('path');
-const childProcess = require('child_process');
+import {Octokit} from '@octokit/rest';
+import {createAppAuth} from '@octokit/auth-app';
+import * as fs from 'fs';
+import {dirname, join, resolve} from 'path';
+import * as childProcess from 'child_process';
+
+interface PullRequest {
+  owner: string;
+  repo: string;
+  number: number;
+}
 
 // TODO: this is a URL; split into owner, repo, and PR
 // const prNumber = process.env.AUTORELEASE_PR;
 // https://github.com/googleapis/releasetool/blob/master/releasetool/commands/publish_reporter.sh
-function parseURL (prURL) {
+export function parseURL(prURL: string): PullRequest | undefined {
   // looking for URLs that match this pattern: https://github.com/googleapis/release-please/pull/707
-  const match = prURL.match(/^https:\/\/github\.com\/(?<owner>.+)\/(?<repo>.+)\/pull\/(?<number>\d+)$/);
-  if (match) {
+  const match = prURL.match(
+    /^https:\/\/github\.com\/(?<owner>.+)\/(?<repo>.+)\/pull\/(?<number>\d+)$/
+  );
+  if (match?.groups) {
     return {
       owner: match.groups.owner,
       repo: match.groups.repo,
-      number: Number(match.groups.number)
+      number: Number(match.groups.number),
     };
   }
+  return undefined;
 }
 
-function getOctokitInstance () {
-  const appId = readFileSync(process.env.APP_ID_PATH, 'utf8');
-  const privateKey = readFileSync(process.env.GITHUB_PRIVATE_KEY_PATH, 'utf8');
-  const installationId = readFileSync(process.env.INSTALLATION_ID_PATH, 'utf8');
+export function getOctokitInstance(
+  appIdPath: string,
+  privateKeyPath: string,
+  installationIdPath: string
```

Comment:
change: made these explicit - read the env vars at the edge of the code (the bin script)